### ISSUE TITLE
The unpool is not standard right ?

### ISSUE THREAD

**Issue by DrM175u** [andreaazzini/segnet#9]
Hello, 

from the code I have read in your project, it the unpool operation is just an image resize. So basically, this 
network is not really a segnet network (at least not in a 'canonical' way from the paper).
Do you plan to use the locations of the max-pool layers to implement the standard unpool op ?

**Comment by jeroen-rooijmans** [comment]
@DrM175u I am currently working on this, there is also some discussion here: https://github.com/tensorflow/tensorflow/issues/2169. I will keep you up to date if I successfully implement the unpooling using the max indices. Note that you have to register the MaxPool gradient to MaxPoolWithArgmax because that is currently missing in Tensorflow.

**Comment by andreaazzini** [comment]
@DrM175u correct, the network I have implemented is not the canonical segnet. I have implemented the canonical version, but I decided to leave it outside the scope of the project. It was definitely painful to train it using the argmax-based unpooling layers and, as @jeroen-rooijmans pointed out, MaxPoolWithArgmax is not available for CPUs.
